### PR TITLE
Creates CI localizationArtifacts for localization team

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/885

Regression? Last working version: No

## Description

This PR creates a new CI artifact named `localizationArtifact`. This artifacts contains:

- English NuGet binaries: All NuGet binaries that can be localizable.
- Localization metadata files: .lcg (genated in CI), .lci (localization comments), .lcl (localized files).
- LocProject.json file: Contains file entries that describe all files to be localized, contained in the artifact.

**Note**: This PR does not affect current localization workflow via build share.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - ~[ ] Automated tests added~
  - ~[ ] Test exception~
  - [x] N/A: This is a CI change. It does not contain product changes.

- **Validation**: Created a tool named `LocProjectValidator` to verify that `LocProject.json` file entries matches a file inside the localization artifact. The tool will be available in NuGet/Entropy once it's checked in. [PR here](https://github.com/NuGet/Entropy/pull/50). The tool reported 14 missing .lci files; to be addressed in this [issue](https://github.com/NuGet/Client.Engineering/issues/904).

- **Documentation**
  - ~[ ] Documentation PR or issue filled~
  - [x] N/A

### CODE REVIEW

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [inline eng/pipelines/templates/Build_and_UnitTest.yml]: Was this just for debugging? Did you intend to remove it before merging?

Comment (dominoFire) [reply eng/pipelines/templates/Build_and_UnitTest.yml]: Yeah, that was for debugging. Nice catch!

**Review by nexue2020 — APPROVED** [review]

**Review by nexue2020 — APPROVED** [review]
Approve to unblock.

**Review by aortiz-msft — APPROVED** [review]

**Pull request merged**
